R/theme-defaults.r

```diff
@@ -639,4 +645,19 @@ theme_test <- function(base_size = 11, base_family = "",
 
     complete = TRUE
   )
+
+  # make sure all elements are set to NULL if not explicitly defined
+  theme_all_null() %+replace% t
+}
+
+theme_all_null <- function() {
+  # set all elements in the element tree to NULL
+  elements <- sapply(
+    names(ggplot_global$element_tree),
+    function(x) NULL,
+    simplify = FALSE, USE.NAMES = TRUE
+  )
+
+  args <- c(elements, list(complete = TRUE))
+  do.call(theme, args)
 }
```

Comment:
Could we cache this theme as well?

Reply:
Sure, will do that tomorrow.

Reply:
Done. I'm also wondering whether this function should be exported, so external package developers can use the same trick, or whether we should rather encourage them to always start with one of the usable ggplot2 themes.

Reply:
I think it is better to guide them towards modifying an existing theme... They can always base it off `theme_void()` if they want a clean slate

Reply:
Sounds good. In any case, I think we need a vignette or other document to provide best practices recommendations for themes. I'll probably write something like that when I'm done with the theme reworking.

Reply:
I did a quick check. It looks like all the most popular theme packages already always start from an existing theme. In fact, I couldn't find any example that didn't do this.